service/controller/app/resource/configmap/desired.go

```diff
@@ -41,6 +42,19 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 		return configMap, nil
 	}
 
+	// If no user-provided configmap name is present, check if a *-user-values config map exists and set the reference
+	if key.UserConfigMapName(cr) == "" {
+		userCM, err := cc.Clients.K8s.K8sClient().CoreV1().ConfigMaps(r.chartNamespace).Get(ctx, fmt.Sprintf("%s-user-values", cr.Name), metav1.GetOptions{})
+		if err == nil || !apierrors.IsNotFound(err) {
```

Comment:
I am thinking that I would only proceed if there are no errors. In theory any other error occurs - e.g. no permission or context deadline - it would update the CR and promotes these permanent / temporary issues further. Same apply for the secrets part.

Reply:
Hmm... yeah, not sure why I did it like that. You're totally right. Updated :)

Reply:
Cool, thank you! :) 👍🏻